### PR TITLE
Update models docs to include signature inference

### PR DESCRIPTION
**Pull request by jerrylian-db**
## Related Issues/PRs

Builds off of this PR: https://github.com/mlflow/mlflow/pull/8725

## What changes are proposed in this pull request?

This PR improves input example documentation in the models docs and adds content on automatic signature inference.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -352,6 +348,12 @@ by hand or :py:func:`inferred <mlflow.models.infer_signature>` from datasets wit
 (e.g. the training dataset with target column omitted) and valid model outputs (e.g. model
 predictions generated on the training dataset).
 
+You may also include a signature with your model by omitting the :py:class:`signature object
+<mlflow.models.ModelSignature>` from the log_model call and instead passing an :ref:`input example <input-example>`.
+Then, for supported MLflow flavors (``sklearn``, ``xgboost``, ``tensorflow``, ``transformers``, and
+``spark``), log_model will automatically infer the signature from the input example and the model's
```
Comment (harupy) [inline docs/source/models.rst:354]: We only support these flavors?

File: docs/source/models.rst [hunk 2]
```diff
@@ -558,19 +560,27 @@ with a new model signature.
...
 For models accepting column-based inputs, an example can be a single record or a batch of records. The
-sample input can be passed in as a Pandas DataFrame, list or dictionary. The given
-example will be converted to a Pandas DataFrame and then serialized to json using the Pandas split-oriented
+sample input can be in the following formats:
+
+* Pandas DataFrame
+* Python ``dict`` (of scalars, strings, or lists of scalars values)
+* Python ``list``
+* Python ``str``
+* Python ``bytes``
```
Comment (harupy) [inline docs/source/models.rst:581]:
```suggestion
* Pandas DataFrame
* ``dict`` (of scalars, strings, or lists of scalar values)
* ``list``
* ``str``
* ``bytes``
```

File: docs/source/models.rst (hunk 1 above)
Comment (jerrylian-db) [reply docs/source/models.rst:354]: I'm building support for more flavors. I'll update this list as I implement and land more model flavors. This PR captures the flavors supported in https://github.com/mlflow/mlflow/pull/8725

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @jerrylian-db !

**Pull request merged**